File: backend/src/routes/safety.ts
import { Router } from "express";
import { scoreSafety } from "../services/handlers.js";

export const safetyRouter = Router();

safetyRouter.post("/score", async (req, res, next) => {
  try {
    const { lat, lng, radiusMeters = 300 } = req.body ?? {};
    if (typeof lat !== "number" || typeof lng !== "number") {
      res.status(400).json({ error: "lat and lng are required numbers" });
      return;
    }
    const data = await scoreSafety(lat, lng, radiusMeters);
    res.json(data);
  } catch (e) {
    next(e);
  }
});

File: backend/guardian/routers/safety.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from guardian.services.handlers import score_safety

router = APIRouter()


class SafetyScoreBody(BaseModel):
    lat: float
    lng: float
    radiusMeters: float = 300


@router.post("/score")
async def safety_score(body: SafetyScoreBody):
    return await score_safety(body.lat, body.lng, body.radiusMeters)
